packages/test/test-utils/src/loaderContainerTracker.ts

```diff
@@ -0,0 +1,34 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ILoader, IContainer } from "@fluidframework/container-definitions";
+
+export class LoaderContainerTracker {
+    private readonly containers = new Set<IContainer>();
+
+    public add<LoaderType extends ILoader>(loader: LoaderType) {
+        const patch = <T, C extends IContainer>(fn: (...args: T[]) => Promise<C>) => {
+            const boundFn = fn.bind(loader);
+            return async (...args: T[]) => {
+                const container = await boundFn(...args);
+                this.containers.add(container);
+                return container;
+            };
+        };
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        loader.resolve = patch(loader.resolve);
```

Comment:
i've actually been thinking we probably want and even on the loader for everytime a container is created, as i see hosts wanting to hook up logging, error handling, etc on every contianer loaded through their container

Reply:
Yeah.  I was kinda thinking the same.